### DISCUSSION TITLE
Benefits?

### DISCUSSION THREAD

**Discussion by Detrol** [AbdBarho/stable-diffusion-webui-docker#591] [Q&A] [answered]
Hey! I just found this repo, which looks interesting. But i wonder if there are any benefits of running this on docker rather than locally on windows?

**Comment by DevilaN** [comment]
Nothing worth writing again here. Just regular every benefit that using docker gives. Just google for "benefits of using docker" and decide whether this is what you are looking for.

**Reply by Detrol** [reply]
Thank you, but I do use Docker for web development, which makes sense because it can utilize Linux utilities that are limited to that, but I can't make sense of it for stable diffusion. I'm therefor unsure what the benefits would be, since that doesn't seem to be the case with SD.

**Reply by DevilaN** [reply, marked as answer]
Zero cost of reverting to vanilla app after extension made mess with package dependencies.
Nothing to worry about python versions and other packages when you really do use other python apps in your system that requires different package versions.
Easy reverting back to previous version of app if there is bug / something don't work as you wish after version upgrade.
There is almost no gain of not using docker, except for storage space for each docker image and being a bit behind with updates.

There are plenty more things, but it depends on user what benefits are preferred and why.

**Reply by Detrol** [reply]
Thank you, lots of good points i didn't think about.

**Reply by AbdBarho** [reply]
One thing I would like to add for this repo specifically is reproducibility, avoiding the "works on my machine" problem.
